waku/v2/node/wakunode2_test.go

```diff
@@ -318,3 +321,222 @@ func TestDecoupledStoreFromRelay(t *testing.T) {
 	require.Len(t, result.Messages, 1)
 	require.Equal(t, msg.Timestamp, result.Messages[0].Timestamp)
 }
+
+func TestStaticShardingMultipleTopics(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Node1 with Relay
+	hostAddr1, err := net.ResolveTCPAddr("tcp", "0.0.0.0:0")
+	require.NoError(t, err)
+	wakuNode1, err := New(
+		WithHostAddress(hostAddr1),
+		WithWakuRelay(),
+		WithClusterID(uint16(20)),
+	)
+	require.NoError(t, err)
+	err = wakuNode1.Start(ctx)
+	require.NoError(t, err)
+	defer wakuNode1.Stop()
+
+	pubSubTopic1 := protocol.NewStaticShardingPubsubTopic(uint16(21), uint16(0))
```

Comment:
Shouldn't 20 be passed as clusterID as node1 is being initialised with clusterID as 20?

Reply:
It is on purpose to check if sharded cluster ID could be different than native clusterID - message still delivered. I've added variables to display intent. Not sure if "native" and sharded are the good labels. https://github.com/waku-org/go-waku/pull/1060/commits/99e3f334cd8356d343d3f3f7ac750ca46508de05

Reply:
Hmm, that is because clusterID is just a user abstraction for underlying pubsubTopic. 
When we do subscribe for relay, we don't check if the node is initialized with that clusterID.
Not sure if we will add such a check in go-waku, but it would be nice to not allow this.

cc @richard-ramos, something to be considered for nwaku bindings.

Reply:
I wonder what is the story behind the lonely clusterID ?  Topic is the central concept we use to shard the entire network isn't it ? Why do we need on the side clusterID which is just one of many possible topics ?

Reply:
That is to help with discovery of peers. If we want to scale beyond 1 pubsubTopic, we can do that by having multiple pubsubTopics. But, how do we discover peers in the network then?
We use discv5 for discovery which depends on ENR. ENR has a size restriction where-in we can't include a list of pubsubTopics in it. 

Hence to make it easier to encode this information in the ENR, the pubsubTopic has been formatted into clusterID and shards. 
The idea of clusterID is kind of similar to networkID in a blockchain , where-in a node is expected to be part of 1 cluster only.
You can refer to rfc https://github.com/waku-org/specs/blob/master/standards/core/relay-sharding.md for more details. 

The limit of 1024 shards within a cluster also has similar reasons behind it.  One of them being discv5 DHT not being able to scale beyond a particular number of nodes i.e finding nodes beyond a particular size is going to take a lot of time that it becomes almost useless. IIRC that node count is 10K or something like that. 


Hope this helps.

Reply:
Thanks for explanation.

Reply:
I've unified clusterID with pubsub topic at https://github.com/waku-org/go-waku/pull/1060/commits/dd960ef93dc1c72149e3561d326c77f4721b57b4